File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mp;

        for(int i = 0 ; i < nums.size(); i++){
            int curr = target - nums[i]; 
            if(mp.find(curr) != mp.end()){
                return {mp[curr] , i};
            }
            mp[nums[i]] = i;
        }
    return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numMap = {}
        for i in range(len(nums)):
            curr = target - nums[i]
            if curr in numMap:
                return [i,numMap[curr]]
            numMap[nums[i]] = i            

        return -1